src/sas/qtgui/Perspectives/Corfunc/RealSpaceCanvas.py

```diff
@@ -44,12 +23,13 @@ def draw_data(self):
         self.axes.set_title("Real Space Correlations")
         self.fig.tight_layout()
 
-        if self.data:
-            data1, data3, data_idf = self.data
-            self.axes.plot(data1.x, data1.y, label="1D Correlation")
-            self.axes.plot(data3.x, data3.y, label="3D Correlation")
-            self.axes.set_xlim(0, max(data1.x) / 4)
-            self.legend = self.axes.legend()
+        if self.data is not None:
+            if len(self.data) == 2:
```

Comment:
can be squeezed into one line

Reply:
Its a suggestion in the same way that a police officer might suggest that you've had too much to drink and that its time to go home.